pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -204,18 +204,34 @@ func (c *NPLController) enqueueSvcUpdate(oldObj, newObj interface{}) {
 	}
 
 	podKeys := sets.String{}
+	var oldPodSet, newPodSet sets.String
 	if oldSvcAnnotation != newSvcAnnotation {
 		// Process Pods corresponding to Service with valid NPL annotation.
 		if oldSvcAnnotation == "true" {
-			podKeys = sets.NewString(c.getPodsFromService(oldSvc)...)
+			oldPodSet = sets.NewString(c.getPodsFromService(oldSvc)...)
+			podKeys = oldPodSet
 		} else if newSvcAnnotation == "true" {
-			podKeys = sets.NewString(c.getPodsFromService(newSvc)...)
+			newPodSet = sets.NewString(c.getPodsFromService(newSvc)...)
+			podKeys = newPodSet
 		}
-	} else if !reflect.DeepEqual(oldSvc.Spec.Selector, newSvc.Spec.Selector) {
+	}
+
+	if oldSvc.Spec.Type != newSvc.Spec.Type {
```

Comment:
Should we check one of old or new type is NodePort here, if we just care about changing to NodePort or changing from NodePort?